src/settings/accounts-settings.test.tsx

```diff
@@ -1328,6 +1328,94 @@ describe('Account setting', () => {
 		).toBeVisible();
 	});
 
+	test('When modify an identity, should populate a ModifyIdentityRequest', async () => {
+		const defaultFirstName = faker.person.firstName();
+		const defaultLastName = faker.person.lastName();
+		const defaultFullName = faker.person.fullName({
+			firstName: defaultFirstName,
+			lastName: defaultLastName
+		});
+		const defaultEmail = faker.internet.email({
+			firstName: defaultFirstName,
+			lastName: defaultLastName
+		});
+		const defaultId = faker.string.uuid();
+
+		const identitiesArray: Array<Identity> = [
+			{
+				id: defaultId,
+				name: 'DEFAULT',
+				_attrs: {
+					zimbraPrefIdentityName: defaultFullName,
+					zimbraPrefReplyToEnabled: 'FALSE',
+					zimbraPrefFromDisplay: defaultFirstName,
+					zimbraPrefFromAddress: defaultEmail,
+					zimbraPrefIdentityId: defaultId,
+					zimbraPrefFromAddressType: 'sendAs'
+				}
+			}
+		];
+
+		const account: Account = {
+			name: defaultEmail,
+			rights: { targets: [] },
+			signatures: { signature: [] },
+			id: defaultId,
+			displayName: '',
+			identities: {
+				identity: identitiesArray
+			}
+		};
+		useAccountStore.setState((previousState) => ({
+			...previousState,
+			account,
+			settings: {
+				...previousState.settings,
+				attrs: {
+					...previousState.settings.attrs,
+					zimbraIdentityMaxNumEntries: 20
+				}
+			}
+		}));
+
+		server.use(
+			rest.post('/service/soap/BatchRequest', (req, res, ctx) =>
+				res(
+					ctx.json({
+						Body: {
+							BatchResponse: {
+								ModifyPrefsResponse: [{ _jsns: 'urn:zimbraAccount' }]
+							}
+						}
+					})
+				)
+			)
+		);
+
+		const batchRequestUrl = '/service/soap/BatchRequest';
+		const pendingBatchRequest = waitForRequest('POST', batchRequestUrl);
+		const { user } = setup(<AccountsSettings />);
+		await waitForGetRightsRequest();
+
+		const accountNameInput = screen.getByRole('textbox', { name: /account name/i });
+
+		const newName = 'Updated Name';
+		await user.clear(accountNameInput);
+		await user.type(accountNameInput, newName);
+
+		await user.click(screen.getByRole('button', { name: /save/i }));
+
+		const request = await pendingBatchRequest;
+
+		const requestBody = (request?.body as { Body: { BatchRequest: BatchRequest } }).Body;
```

Comment:
request.body is deprecated, I think we can replace it with request.json()
```suggestion
		const { Body: requestBody } = await pendingBatchRequest.then((req) =>
			req.json<{ Body: { BatchRequest: BatchRequest } }>()
		);
```